### PR TITLE
fix: add back client constructors with client options and actually use its reliability settings

### PR DESCRIPTION
**Pull request by childish-sambino**
Fixes https://github.com/sendgrid/sendgrid-csharp/issues/1000
Relates to https://github.com/sendgrid/sendgrid-csharp/pull/839

When the default client options were made static and the constructors later removed, there was no way to modify the reliability settings used by the retry handler. This fix adds back the constructors so clients can now be constructed with just the client options and the options will be used when creating a retry handler, if any.

### CODE REVIEW

File: src/SendGrid/BaseClient.cs [hunk 1]
```diff
@@ -42,7 +42,7 @@ public abstract class BaseClient : ISendGridClient
         /// </summary>
         /// <param name="options">A <see cref="BaseClientOptions"/> instance that defines the configuration settings to use with the client.</param>
         /// <returns>Interface to the Twilio SendGrid REST API.</returns>
-        public BaseClient(BaseClientOptions options)
+        protected BaseClient(BaseClientOptions options)
```
Comment (eshanholtz) [inline src/SendGrid/BaseClient.cs:45]: Why the change to protected?

Comment (childish-sambino) [reply src/SendGrid/BaseClient.cs:45]: In C# there's no such thing as public abstract base constructors. This change just makes that fact explicit.

Comment (eshanholtz) [reply src/SendGrid/BaseClient.cs:45]: Ah, missed the abstract part 👍
